### PR TITLE
Adds zombie removal tool

### PR DESCRIPTION
**Pull request by supertopher**
## What
We had zombies

## How
kills zombies by identifying runners not currently active and shooting them

closes cloud issue https://github.com/airbytehq/airbyte-cloud/issues/1840

### CODE REVIEW

File: tools/bin/gh_action_zombie_killer [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  echo -e "\033[31m\nFAILED TO SOURCE TEST RUNNING OPTIONS.\033[39m"
+  echo -e "\033[31mTried $relative_path_to_defaults\033[39m"
+  exit 1
+fi
+
+echo "To run locally use GITHUB_PAT=\$YOUR_PAT_HERE before running"
+token=$GITHUB_PAT
+org=airbytehq
+# FUN POSIX fact, every string is an array!
+repo_list="airbyte airbyte-cloud"
```
Comment (davinchia) [inline tools/bin/gh_action_zombie_killer:24]: commenting to make sure the github service account has permissions to airbyte cloud repo

File: tools/bin/gh_action_zombie_killer [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  cursor=0
+  for this_runner in $all_runner_ids; do
+    cursor=$(($cursor+1))
+    echo -e "$blue_text""Removing $cursor / $all_runner_ids_count""$default_text"
+    curl \
+      --silent \
+      --request DELETE \
+      --header "Accept: application/vnd.github.v3+json" \
+      --header "Authorization: token $token" \
+      https://api.github.com/repos/$org/$repo/actions/runners/$this_runner && \
```
Comment (davinchia) [inline tools/bin/gh_action_zombie_killer:68]: nit: drop api docs link

File: tools/bin/gh_action_zombie_killer [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (davinchia) [reply tools/bin/gh_action_zombie_killer:1]: agree. little value is making this pretty as long as all the context is captured in script or PR.

File: .github/workflows/terminate-zombie-build-instances.yml [hunk 4]
```diff
@@ -34,3 +34,9 @@ jobs:
 
           # See https://docs.aws.amazon.com/cli/latest/reference/ec2/terminate-instances.html for terminate command.
           echo $to_terminate |  jq '.[]  | .InstanceId' | xargs --no-run-if-empty --max-args=1 aws ec2 terminate-instances --instance-ids
+
+    steps:
+      - shell: List and Terminate GH actions runners Older than One Day
```
Comment (davinchia) [inline .github/workflows/terminate-zombie-build-instances.yml:39]: I don't see logic to filter out runners older than a day. Am I missing something?

File: tools/bin/gh_action_zombie_killer [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      --silent \
+      --header "Accept: application/vnd.github.v3+json" \
+      --header "Authorization: token $token" \
+      --request GET  https://api.github.com/repos/$org/$repo/actions/runners?page=$page_count&per_page=100)
+    runner_response_wc=$(echo $runner_response | wc -w)
+    # For auth errors
+    if test $runner_response_wc -lt 100; then
+      echo -e "$blue_text""\$runner_response is \n\n$runner_response\n\n""$default_text"
+    fi
+
```
Comment (git-phu) [inline tools/bin/gh_action_zombie_killer:48]: Are there other possible statuses (like starting up or shutting down) that we don't want to delete?

**Review by git-phu — APPROVED** [review]
Looks good!

Had one question

File: .github/workflows/terminate-zombie-build-instances.yml (hunk 4 above)
Comment (supertopher) [reply .github/workflows/terminate-zombie-build-instances.yml:39]: that may have been an aspirational day when I started writing the script, I'll update the comment.  

We are deleting anything not "online" but I'm changing that in an upcoming PR to `==` offline which seems to be the end state of these runners.  Happy to add day logic though

Comment (davinchia) [reply .github/workflows/terminate-zombie-build-instances.yml:39]: no need. wanted to make sure I understood the logic. I think filtering for offline makes sense. I believe the action status is online -> offline. would double check that runners don't start in offline.

**Review by davinchia — APPROVED** [review]
Thanks!

Main comment here is to update the step title and to double check `offline` status is correct. Otherwise approving since I'm about to go offline and don't want to block.

File: tools/bin/gh_action_zombie_killer (hunk 5 above)
Comment (supertopher) [reply tools/bin/gh_action_zombie_killer:48]: there are, I changed the the filter to find `offline`

"Active", "Offline",  and "Online" are the three regular statuses

**Comment by 2370131441** [comment]
An exception occurred applying plugin request [id: 'airbyte-docker']
> Failed to apply plugin 'airbyte-docker'.
   > A problem occurred starting process 'command 'docker'';

Why is that?

**Pull request merged**